core/include/scipp/core/dtype.h

```diff
@@ -8,8 +8,8 @@
 #include <Eigen/Dense>
 #include <boost/container/small_vector.hpp>
 
-namespace pybind11 {
-class object;
+namespace scipp::python {
+class PyObject;
```

Comment:
We still forward declare a type that is part of a dependent library. I recall this causing problems with type lists.

Reply:
Could you clarify what you mean by "type lists"?

Reply:
Anywhere you declare a std::variant. For example https://github.com/scipp/scipp/blob/master/core/include/scipp/core/variable.h#L48. I can't remember exactly where the problem cropped up before, but I think it might have been at the point we needed binary operators over variables containing pyobjects.

Reply:
I do not intend to add such operators, but you are right that we cannot use it with `transform`.